### PR TITLE
Update event

### PR DESCRIPTION
**Pull request by Tiffany-Zhu5303**
Updating event endpoint done

### CODE REVIEW

File: handlers/events.go [hunk 1]
```diff
@@ -37,18 +38,102 @@ func GetEvent(c *gin.Context) {
...
 	fmt.Printf("%+v", eventrow)
-	
+
 	for eventrow.Next() {
 		var event models.GetEvent
-        // for each row, scan into the event struct
+		// for each row, scan into the event struct
 		err = eventrow.Scan(&event.EventID, &event.EventTitle, &event.Date, &event.Time, &event.Location, &event.HostName, &event.Description, &event.ContactInfo, &event.PublicPrivate, &event.NumRSVP, &event.MaxAttendees)
 		if err != nil {
+			fmt.Println(err)
```
Comment (dianaabishop) [inline handlers/events.go:47]: stray logging that you should be able to remove now! 🎉

File: handlers/events.go [hunk 2]
```diff
@@ -37,18 +38,102 @@ func GetEvent(c *gin.Context) {
 		see_row)
 
 	fmt.Printf("%+v", eventrow)
```
Comment (dianaabishop) [inline handlers/events.go:40]: You didn't add this one, but while you're in here maybe you can delete this stray logging too 😄

File: handlers/events.go [hunk 3]
```diff
@@ -37,18 +38,102 @@ func GetEvent(c *gin.Context) {
...
+
+	eventInfo, err := dbmap.Query(
+		"SELECT event_id, title, date, time, location, host_name, description, contact_info, public_private, num_of_RSVP, max_attendees FROM event WHERE event_id=?;", id)
+
+	for eventInfo.Next() {
+		var event models.GetEvent
+		// for each row, scan into the event struct
+		err = eventInfo.Scan(&event.EventID, &event.EventTitle, &event.Date, &event.Time, &event.Location, &event.HostName, &event.Description, &event.ContactInfo, &event.PublicPrivate, &event.NumRSVP, &event.MaxAttendees)
+		if err != nil {
+			fmt.Println(err)
```
Comment (dianaabishop) [inline handlers/events.go:72]: stray logging here ✂️

File: handlers/events.go [hunk 4]
```diff
@@ -37,18 +38,102 @@ func GetEvent(c *gin.Context) {
...
+			c.IndentedJSON(http.StatusInternalServerError, nil) //server error
+			return
+		}
+		// append the event into events array
+		savedEvent = append(savedEvent, event)
+	}
+
+	fmt.Printf("%+v", savedEvent)
```
Comment (dianaabishop) [inline handlers/events.go:80]: and here ✂️

File: handlers/events.go [hunk 5]
```diff
@@ -37,18 +38,102 @@ func GetEvent(c *gin.Context) {
...
+	if updateEvent.MaxAttendees == 0 {
+		updateEvent.MaxAttendees = savedEvent[0].MaxAttendees
+	}
+
+	_, err = dbmap.Query(
+		"UPDATE event SET title = ?, date = ?, time = ?, location = ?, host_name = ?, description = ?, contact_info = ?, public_private = ?, num_of_RSVP = ?, max_attendees = ? WHERE event_id = ?",
+		updateEvent.EventTitle, updateEvent.Date, updateEvent.Time, updateEvent.Location, updateEvent.HostName, updateEvent.Description, updateEvent.ContactInfo, updateEvent.PublicPrivate, updateEvent.NumRSVP, updateEvent.MaxAttendees, id)
+
+	if err != nil {
 		fmt.Println(err)
```
Comment (dianaabishop) [inline handlers/events.go:133]: and here even though you didn't add it 😄 ✂️

File: handlers/events.go [hunk 6]
```diff
@@ -37,18 +38,102 @@ func GetEvent(c *gin.Context) {
...
+
+	fmt.Printf("%+v", savedEvent)
+
+	// Call BindJSON to bind the received JSON to event +add error handling later
+	if err := c.BindJSON(&updateEvent); err != nil {
+		c.IndentedJSON(http.StatusBadRequest, nil) //bad data
+		return
+	}
+
+	if updateEvent.EventTitle == "" {
```
Comment (dianaabishop) [inline handlers/events.go:88]: woooo!! Update endpoints are always the hardest - awesome job catching all these cases below!

**Review by dianaabishop — APPROVED** [review]
AMAZING job!!!! All looks great, just a few stylistic/cleanup things - otherwise 💯

File: handlers/events.go (hunk 1 above)
Comment (Tiffany-Zhu5303) [reply handlers/events.go:47]: done!

File: handlers/events.go (hunk 2 above)
Comment (Tiffany-Zhu5303) [reply handlers/events.go:40]: I removed it.

File: handlers/events.go (hunk 3 above)
Comment (Tiffany-Zhu5303) [reply handlers/events.go:72]: done!

File: handlers/events.go (hunk 4 above)
Comment (Tiffany-Zhu5303) [reply handlers/events.go:80]: All removed.

File: handlers/events.go (hunk 5 above)
Comment (Tiffany-Zhu5303) [reply handlers/events.go:133]: done!

File: handlers/events.go [hunk 7]
```diff
@@ -37,18 +38,102 @@ func GetEvent(c *gin.Context) {
...
 		fmt.Println(err)
 		c.IndentedJSON(http.StatusInternalServerError, nil) //server error
 		return
 	}
-        // append the event into events array
-		events = append(events, event)
-	}
-	c.JSON(201, events) //success
+
+	c.JSON(201, updateEvent) //success
```
[the remark this answers is not in the document]
Comment (Tiffany-Zhu5303) [reply handlers/events.go:138]: I changed it to 200.

**Comment by sharronq** [comment]
Tested successfully, ready for merge

**Pull request merged**